lib/monitor.js

```diff
@@ -0,0 +1,43 @@
+'use babel'
+
+import log from './log'
+import commandLog from './command-log'
+
+function hostIP() {
+  return LEARN_IDE_HOST_IP
+}
+
+function clearHostIP() {
+  LEARN_IDE_HOST_IP = null
+}
+
+export default function monitor(terminal, subscriptions) {
+  subscriptions.add(atom.commands.onDidDispatch(({type}) => {
+    if (type === 'application:quit') { log.add({event: 'quit'}) }
```

Comment:
I'm not sure that we can guarantee it; however, it's been hitting this really reliably. We can move it up a bit using `atom.commands.onWillDispatch`, though I dunno how much that actually buys us.